lib/src/function-registry.ts

```diff
@@ -119,3 +126,46 @@ export class FunctionRegistry<sync extends 'sync' | 'async'> {
     }
   }
 }
+
+/**
+ * Implements the simplification algorithm for custom function return values.
+ *  {@link https://github.com/sass/sass/blob/main/spec/types/calculation.md#simplifying-a-calculationvalue}
+ */
+function simplify(value: unknown): unknown {
+  if (value instanceof SassCalculation) {
+    const simplifiedArgs = value.arguments.map(
+      simplify
+    ) as List<CalculationValue>;
+    if (value.name === 'calc') {
+      if (simplifiedArgs.size !== 1) {
+        throw new Error('calc() requires exactly 1 argument.');
+      }
+      return simplifiedArgs.get(0);
+    }
+    if (value.name === 'clamp') {
+      if (simplifiedArgs.size !== 3) {
+        throw new Error('clamp() requires exactly 3 arguments.');
+      }
+      return SassCalculation.clamp(
+        simplifiedArgs.get(0) as CalculationValue,
+        simplifiedArgs.get(1),
+        simplifiedArgs.get(2)
+      );
+    }
+    if (value.name === 'min') {
+      return SassCalculation.min(simplifiedArgs);
+    }
+    if (value.name === 'max') {
+      return SassCalculation.max(simplifiedArgs);
+    }
+    throw new Error(`Unknown calculation function: ${value.name}.`);
+  }
+  if (value instanceof CalculationOperation) {
+    return new CalculationOperation(
```

Comment:
@nex3 in the [Dart implementation](https://github.com/oddbird/dart-sass/blob/e70e5e39f029d99d77397030988d7e6ee384acfb/lib/src/node/compile.dart#L248) of `simplify`, we can call `SassCalculation.operate` to collapse `CalculationOperation`s into plain values, but that method is not available in the host. Since we want to leverage as much as the Dart logic as possible, I'm hesitant to implement [`_operateInternal`](https://github.com/oddbird/dart-sass/blob/e70e5e39f029d99d77397030988d7e6ee384acfb/lib/src/value/calculation.dart#L182) in TS. Do you see another viable alternative?

Reply:
The expectation is that the host returns an _entirely unsimplified_ calculation AST to the compiler. The simplification will be handled by the compiler on its own. You probably don't need a `simplify()` function at all.

Reply:
Ah, sorry, I was way overcomplicating it 😅 Should be fixed after 8d2d5a0193238d16c7c0bd4f7a4192ff8de4f1c9